src/payment/checkout/payment-form/flex-microform/FlexMicroform.jsx

```diff
@@ -15,20 +15,21 @@ class FlexMicroform extends React.Component {
 
     window.microform = null;
     this.state = {
-      microformStatus: DEFAULT_STATUS,
+      captureKey: { microformStatus: DEFAULT_STATUS },
     };
   }
 
   componentDidMount() {
     this.initialize();
   }
 
-  componentDidUpdate() {
-    this.initialize();
+  componentDidUpdate(prevProps) {
+    this.initialize(prevProps);
   }
 
-  initialize = () => {
-    if (window.microform !== null || !this.props.captureKeyId) {
+  initialize = (prevProps = {}) => {
+    console.log("FlexMicroform Init");
+    if (!this.props.captureKeyId || prevProps.captureKeyId === this.props.captureKeyId) {
```

Comment:
The `initialize` arguments and prop checking here may be more well-factored as:

```
componentDidMount() {
    this.initialize(this.props.captureKeyId);
}

componentDidUpdate(prevProps) {
  if (prevProps.captureKeyId !== this.props.captureKeyId) {
    this.initialize(prevProps.captureKeyId);
  }
}

initialize = (captureKeyId) => {
  // ... existing contents minus the short circuiting conditional...
}
```

That way the "Should we do this thing?" logic is contained in componentDidUpdate, where it naturally fits, and the initialize function itself just takes the one prop it actually needs (`captureKeyId`) and can just assume it's correct, and doesn't need to care if it's from the current or previous props.